packages/ember-routing/lib/location/history_location.js

```diff
@@ -150,7 +151,8 @@ export default EmberObject.extend({
    @param path {String}
   */
   pushState(path) {
-    let state = { path: path };
+    let id = guidFor(`${path}-${new Date()}`);
```

Comment:
I don't think using `guidFor` on a string is very useful, if the idea is that all new states have a unique `id` I think this may be easier/better:

```js
let state = { path, id: ++stateCounter };
```